src/CommunityToolkit.Maui/Controls/Avatar.cs

```diff
@@ -0,0 +1,557 @@
+using System.ComponentModel;
+using Microsoft.Maui.Controls.Internals;
+using Microsoft.Maui.Controls.Shapes;
+
+namespace CommunityToolkit.Maui.Controls;
+
+/// <summary>Avatar content view.</summary>
+public class Avatar : ContentView, IAvatarElement, IFontElement, ITextElement, IImageElement
+{
+	/// <summary>The backing store for the <see cref="IAvatarElement.AvatarBackgroundColor" /> bindable property.</summary>
+	public static readonly BindableProperty AvatarBackgroundColorProperty = BindableProperty.Create(nameof(IAvatarElement.AvatarBackgroundColor), typeof(Color), typeof(IAvatarElement), null, propertyChanged: OnBackgroundColorChanged);
+
+	/// <summary>The backing store for the <see cref="IAvatarElement.AvatarHeightRequest" /> bindable property.</summary>
+	public static readonly BindableProperty AvatarHeightRequestProperty = BindableProperty.Create(nameof(IAvatarElement.AvatarHeightRequest), typeof(double), typeof(VisualElement), defaultValue: AvatarElement.DefaultHeightRequest, propertyChanged: OnHeightRequestPropertyChanged);
+
+	/// <inheritdoc/>
+	public static readonly BindableProperty AvatarPaddingBottomProperty = BindableProperty.Create("AvatarPaddingBottom", typeof(double), typeof(IPaddingElement), default(double), propertyChanged: OnPaddingBottomChanged);
+
+	/// <inheritdoc/>
+	public static readonly BindableProperty AvatarPaddingLeftProperty = BindableProperty.Create("AvatarPaddingLeft", typeof(double), typeof(IPaddingElement), default(double), propertyChanged: OnPaddingLeftChanged);
+
+	/// <summary>The backing store for the <see cref="AvatarPadding" /> bindable property.</summary>
+	public static readonly BindableProperty AvatarPaddingProperty = BindableProperty.Create(nameof(IPaddingElement.Padding), typeof(Thickness), typeof(IPaddingElement), default(Thickness), propertyChanged: OnPaddingPropertyChanged, defaultValueCreator: PaddingDefaultValueCreator);
+
+	/// <inheritdoc/>
+	public static readonly BindableProperty AvatarPaddingRightProperty = BindableProperty.Create("AvatarPaddingRight", typeof(double), typeof(IPaddingElement), default(double), propertyChanged: OnPaddingRightChanged);
+
+	/// <inheritdoc/>
+	public static readonly BindableProperty AvatarPaddingTopProperty = BindableProperty.Create("AvatarPaddingTop", typeof(double), typeof(IPaddingElement), default(double), propertyChanged: OnPaddingTopChanged);
+
+	/// <summary>The backing store for the <see cref="AvatarShadow" /> bindable property.</summary>
+	public static readonly BindableProperty AvatarShadowProperty = BindableProperty.Create(nameof(IAvatarElement.AvatarShadow), typeof(Shadow), typeof(VisualElement), defaultValue: null, propertyChanged: OnShadowPropertyChanged);
```

Comment:
You specify the default value of `AvatarShadow` as `null` here, but `public Shadow AvatarShadow` is not nullable.

Should the property be nullable, or should its default value be something other than `null`?

Reply:
I took this from Microsoft.Maui.Controls.VisualElement.impl.cs.  I know that the M$FT code is subject to change, but I thought it best to use exactly as implemented by this repository.  Happy to go in whatever direction you feel is appropriate.

Reply:
AvatarShadow property has been removed.